src/lib/ts/downlevel-transformer.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  ModuleKind,
+  ScriptTarget,
+  createCompilerHost,
+  CompilerOptions,
+  createProgram,
+  formatDiagnostics,
+  transpileModule
+} from 'typescript';
+import * as path from 'path';
+
+import * as log from '../util/log';
+
+/**
+ * Base `tsc` `CompilerOptions` shared among various downleveling methods.
+ */
+const COMPILER_OPTIONS: CompilerOptions = {
+  target: ScriptTarget.ES5,
+  module: ModuleKind.ES2015,
+  allowJs: true,
+  sourceMap: true,
+  importHelpers: true,
+  downlevelIteration: true
+};
+
+/**
+ * Downlevels a .js file from `ES2015` to `ES5`. Internally, uses `tsc`.
+ *
+ */
+export function downlevelEmitWithTsc(entryPoint: string, outDir: string): Promise<void> {
+  log.debug(`tsc downlevel ${entryPoint}`);
+
+  const compilerOptions: CompilerOptions = {
+    ...COMPILER_OPTIONS,
+    mapRoot: path.dirname(entryPoint),
+    outDir
+  };
+
+  const compilerHost = createCompilerHost(compilerOptions);
+  const program = createProgram([entryPoint], compilerOptions, compilerHost);
+  const emitResult = program.emit();
+
+  return emitResult.emitSkipped
+    ? Promise.reject(new Error(formatDiagnostics(emitResult.diagnostics, compilerHost)))
+    : Promise.resolve();
+}
+
+/**
+ * Downlevels a .js file from `ES2015` to `ES5`. Internally, uses `tsc`.
+ *
+ * Required for some external as they contains `ES2015` syntax such as `const` and `let`
+ */
+export function downlevelCodeWithTsc(code: string, filePath: string): Promise<{ code: string; map: string }> {
```

Comment:
Why two different downlevel methods? I didn't spot it.

Reply:
One is required for rollup, which downlevel a single file at a time. The other one is used to downlevel ESM2015 to ESM5 which downlevels all files from an entry point, using a `program`

Reply:
Ok, why? ngc compiles to esm2015. Then, we donwlevel to esm5. FESM15 takes esm2015 as input. FESM5 takes esm5 as input. Why another downlevelling?

Reply:
This is needed for UMD bundles, when we do embedding of external dependencies. This is due the fact that if we are embedding something with ES6 syntax we need to downlevel it. As Rollup won't do that. 

More info
https://github.com/rollup/rollup/issues/2093

Reply:
In all fairness I am considering form es2015 to es5 rather than doenleving Use ngc directly. This will also enable us to have inlinedSourceMaps for es